client/lib/signup/dependency-store.js

```diff
@@ -31,8 +31,8 @@ const SignupDependencyStore = {
 };
 
 function assertValidDependencies( action ) {
-	const providesDependencies = steps[ action.data.stepName ].providesDependencies || [],
-		extraDependencies = difference( keys( action.providedDependencies ), providesDependencies );
+	const providesDependencies = get( steps, [ action.data.stepName, 'providesDependencies' ], [] );
```

Comment:
this has slightly different semantics than the old version. If `providesDependencies` exists, but is null, it will no longer default to `[]`.  See http://jsbin.com/wixexidito/edit?js,console

Reply:
that may be ok. It just depends on what providesDependencies is when it's empty.